client/src/components/EventPage/EventPage.js

```diff
@@ -1,98 +1,135 @@
+   ExpansionPanelDetails,
+   ExpansionPanelSummary
 } from '@material-ui/core';
 import {
+    Face,
+    Info,
+    Comment,
+    Collections,
     Add,
-    DateRange, 
+    DateRange,
     LocationOn,
     ExpandMore
 } from '@material-ui/icons';
 import Swiper from 'react-id-swiper/lib/ReactIdSwiper.full';
 import { Pagination, Autoplay } from 'swiper/dist/js/swiper.esm';
+import Gallery from 'react-grid-gallery';
 import "react-id-swiper/src/styles/scss/swiper.scss";
 
-// Global constants;
-
 // Swipers params for event page;
-const galleryParams = {
+const userParams = {
     modules: [Pagination, Autoplay],
-    effect: 'slide',
-    slidesPerView: 'auto',
-    centeredSlides: false,
-    loop: true,
-    containerClass: 'swiper-container  event-page__gallery-swiper',
+    slidesPerView: 2,
+    slidesPerColumn: 3,
+    containerClass: 'swiper-container  event-page__users-swiper',
     rebuildOnUpdate: true,
     shouldSwiperUpdate: true,
-    pagination: {
-      el: ".swiper-pagination",
-      type: "bullets",
-      clickable: true
+    nested: true,
+    autoplay: {
+        delay: 4000,
+        disableOnInteraction: true,
     },
-    spaceBetween: 16,
-    autoplay: true
+    simulateTouch: false,
+    breakpointsInverse: true,
+    breakpoints: {
+        768: {
+            slidesPerView: 3,
+            slidesPerColumn: 2
+        },
+        1168: {
+            slidesPerView: 2,
+            slidesPerColumn: 2
+        }
+    },
+    pagination: {
+        el: ".swiper-pagination",
+        type: "bullets",
+        clickable: true
+    }
 };
-
 const tabsParams = {
     modules: [Pagination],
-    slidesPerView: 'auto',
+    slidesPerView: 1,
     loop: false,
-    pagination: {
-      el: ".event-page__section-pagination",
-      type: "bullets",
-      clickable: true
-    },
     centeredSlides: true,
     autoHeight: true,
-    spaceBetween: 30,
+    spaceBetween: 0,
     rebuildOnUpdate: true,
     shouldSwiperUpdate: true,
+    noSwipingClass: 'event-page__users-swiper',
     containerClass: 'swiper-container  event-page__tabs-swiper',
-    // Update height;
+    breakpointsInverse: true,
+    breakpoints: {
+        768: {
+            noSwipingClass: 'swiper-container'
+        }
+    },
     on: {
-        click: function() {
-            this.updateAutoHeight();
+        slideChangeTransitionEnd: function() {
+            window.dispatchEvent(new Event('resize'));
+        },
+        resize: function() {
+            this.update();   
         }
     }
-} 
+};
+let swiperInstance;
 
 class EventPage extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            gallery: [],
             title: '',
+            cover: '',
             description: '',
-            users: []
+            users: [],
+            swiper: null,
+            activeSlide: 0,
+            gallery: []
         };
-        this.componentDidMount = this.componentDidMount.bind(this);
+    }
```

Comment:
class EventPage extends Component {
    constructor(props) {
        super(props);
        this.state = {	
            title: '',	          
            cover: '',
            description: '',	           
            users: []	         
            swiper: null,
            activeSlide: 0,
            gallery: []
        };
}
This is the same code but easier for reading:
class EventPage extends Component {
state = {
	      title: '',	          
            cover: '',
            description: '',	           
            users: []	         
            swiper: null,
            activeSlide: 0,
            gallery: [],
}

Reply:
you mean just state, without this keyword and no constructor?

Reply:
Correct! It is old style code, in future you will able to use just "state="

Reply:
Ok thanks, didn't know